src/MSLGenerator.h

```diff
@@ -88,10 +88,6 @@ class MSLGenerator
     Target          m_target;
     
     bool            m_error;
-    
-    // TODO: Temporary hack to access uniforms struct members
-    HLSLBuffer* m_per_pass_buffer;
```

Comment:
What about a 'buffer(name)' semantic for uniforms not explicitly declared inside a buffer? That way the uniform is added to the named buffer or it's automatically created if it does not exist.